src/utils/paths.rs

```diff
@@ -248,19 +248,46 @@ fn get_debuginfod_path(filetype: FileType, identifier: &ObjectId) -> Option<Stri
 }
 
 fn get_unified_path(filetype: FileType, identifier: &ObjectId) -> Option<String> {
+    // determine the suffix and object type
     let suffix = match filetype {
         FileType::ElfCode | FileType::MachCode | FileType::Pe => "executable",
         FileType::ElfDebug | FileType::MachDebug | FileType::Pdb => "debuginfo",
         FileType::Breakpad => "breakpad",
         FileType::SourceBundle => "sourcebundle",
     };
 
-    // for PEs we want to use signature+age that also breakpad uses.  We also use
-    // the debug id if the query did not provide a code id.
-    let id = if filetype == FileType::Pe || identifier.code_id.is_none() {
-        Cow::Owned(identifier.debug_id?.breakpad().to_string().to_lowercase())
-    } else {
-        Cow::Borrowed(identifier.code_id.as_ref()?.as_str())
+    // determine object type.  We prefer to use the file type as indicator for going
+    // back to the object type.  If that is not possible, we use the object type
+    // that is stored on the identifier which might be unreliable.
```

Comment:
It might be worth noting here, that `prepare_download_paths` invokes this function multiple times with different file types. If there is no object type stored on the identifier (which Sentry tries to prevent), then all possible FileTypes are tried out. This means, only source and breakpad lookup will fail, but the other lookups will succeed.